test/e2e/helpers_test.go

```diff
@@ -653,11 +653,21 @@ func runOnlyOnKindClusters(t *testing.T) {
 	t.Helper()
 
 	existingClusterIsKind := strings.Split(existingCluster, ":")[0] == string(kind.KindClusterType)
-	clusterProviderIsKind := clusterProvider == "" || clusterProvider == string(kind.KindClusterType)
+	if existingClusterIsKind {
+		return
+	}
 
-	if !existingClusterIsKind || !clusterProviderIsKind {
-		t.Skip("test is supported only on Kind clusters")
+	clusterProviderIsKind := clusterProvider == string(kind.KindClusterType)
+	if clusterProviderIsKind {
+		return
 	}
+
+	clusterProviderUnspecified := clusterProvider == ""
+	if clusterProviderUnspecified {
+		return
+	}
```

Comment:
I think that won't work as expected in the following situation:

```
KONG_TEST_CLUSTER=gke:name
KONG_TEST_CLUSTER_PROVIDER=
```

Reply:
```suggestion
	clusterProviderUnspecified := clusterProvider == ""
	existingClusterUnspecified := existingCluster == ""
	if clusterProviderUnspecified && existingClusterUnspecified {
		return
	}
```

Reply:
Good catch! Thanks!